ivy/functional/ivy/elementwise.py

```diff
@@ -2473,6 +2473,215 @@ def not_equal(
         an array containing the element-wise results. The returned array must have a
         data type of ``bool``.
 
+    This function conforms to the `Array API Standard
+    <https://data-apis.org/array-api/latest/>`_. This docstring is an extension of the
+    `docstring <https://data-apis.org/array-api/latest/API_specification/generated/signatures.elementwise_functions.not_equal.html>`_ # noqa
+    in the standard.
+
+    Both the description and the type hints above assumes an array input for simplicity,
+    but this function is *nestable*, and therefore also accepts :code:`ivy.Container`
+    instances in place of any of the arguments.
+
+    Functional Examples
+    ------------------
+
+    With :code:`ivy.Array` inputs:
+
+    >>> x1 = ivy.array([1, 0, 1, 1])
+    >>> x2 = ivy.array([1, 0, 0, -1])
+    >>> y = ivy.not_equal(x1, x2)
+    >>> print(y)
+    ivy.array([False, False, True, True])
+
+    >>> x1 = ivy.array([1, 0, 1, 0])
+    >>> x2 = ivy.array([0, 1, 0, 1])
+    >>> y = ivy.not_equal(x1, x2)
+    >>> print(y)
+    ivy.array([True, True, True, True])
+
+    >>> x1 = ivy.array([1, -1, 1, -1])
+    >>> x2 = ivy.array([0, -1, 1, 0])
+    >>> y = ivy.zeros(4)
+    >>> ivy.not_equal(x1, x2, out = y)
+    >>> print(y)
+    ivy.array([1, 0, 0, 1])
```

Comment:
Hi @davidlee1102, another small change to make as we need to keep formatting consistent so that the docstring test can pass:

- [ ] As `ivy.zeros()` is being used here, the data type of the output in line 2507 is a float type so the array should show as `([1., 0., 0., 1.])`
Please update this array.